proto/Dockerfile

```diff
@@ -0,0 +1,24 @@
+# This Dockerfile defines a toolbox for use in linting, formatting, and compiling the Tendermint protos.
+
+FROM bufbuild/buf:latest as buf
+
+FROM alpine:3.14
+
+# Install a commonly used set of programs for use with our protos.
+# clang-extra-tools is included here because it provides clang-format, the program used for formatting the protos.
+RUN echo 'http://dl-cdn.alpinelinux.org/alpine/edge/testing' >> /etc/apk/repositories && \
```

Comment:
I think if we base on `golang:1.17-alpine` we could avoid having to install Go here. More importantly, that would make it clear which version of Go we are getting.

(And with 1.16 or 1.17, we can assume `GO111MODULE=on` by default)